### PR TITLE
fix for push_dir in different OS

### PR DESCRIPTION
**Pull request by tsaridas**
### What does this PR do?
fixes push directory under windows minion and linux master

### What issues does this PR fix or reference?
https://github.com/saltstack/salt/issues/38159

### Previous Behavior
linux master created one directory for each file uploaded 

### New Behavior
new directories are created normally

### Tests written?

No

### CODE REVIEW

**Comment by tsaridas** [comment]
@cachedout can you please take a look at this ? maybe if you agree we can also add it to 2016.11 ?
thanks

File: salt/master.py [hunk 1]
```diff
@@ -1269,9 +1269,12 @@ def _file_recv(self, load):
             return {}
         load.pop('tok')
 
+        # Join path
+        sep_path = os.sep.join(load['path'])
```
Comment (cachedout) [inline salt/master.py]: Was the removal of the `*` to unpack the list an intentional choice here? IIRC, there was a reason we needed this here.

Comment (cachedout) [inline salt/master.py]: But not all minions will be updated at the same time as the master. We need to handle both cases.

Comment (tsaridas) [inline salt/master.py]: you lost me. 
If the tuple is an issue I can try to pass it from the minion in that way but I don't really see the point.
By any means, I'm not trying to disagree just to understand. I'm sure I'm missing some info on how this works.

Comment (cachedout) [inline salt/master.py]: Ah, never mind. Ignore the point about the tuple -- I thought you were expecting a behavior difference but I misunderstood. I see the difference here. There's a behavior difference between `os.sep.join` (which requires an iterable) and `os.path.join` which takes (`path`, `*path`). This approach is fine.

**Comment by rallytime** [comment]
@tsaridas I have back-ported this to 2016.11 with #38203

**Pull request merged**